trunner/runners/common.py

```diff
@@ -305,25 +306,43 @@ def go(self):
         self.plo.send('go!\r\n')
 
 
-class Runner:
+class Runner(ABC):
     """Common interface for test runners"""
 
+    BUSY = 'BUSY'
+    SUCCESS = 'SUCCESS'
+    FAIL = 'FAIL'
+
+    def set_status(self, status):
+        """Method for sygnalising a current runner status: busy/failed/succeeded"""
+        # for now, not used in all target runners
+        self.status = status
+
     def flash(self):
         """Method used for flashing a device with the image containing tests."""
         pass
 
+    @abstractmethod
     def run(self, test):
         """Method used for running a single test case which is represented by TestCase class."""
         pass
 
 
-class DeviceRunner(Runner):
-    """This class provides interface to run test case using serial port"""
+class DeviceRunner(Runner, ABC):
+    """This class provides interface to run tests on hardware targets using serial port"""
 
     def __init__(self, port):
         self.port = port
         self.serial = None
 
+    @abstractmethod
+    def set_status(self, status):
+        pass
+
+    @abstractmethod
```

Comment:
was not abstract in base class (`Runner`). What is happening? :)

Reply:
I thought like that, every DeviceRunner has to have flash() and set_status() methods defined (so they are abstract methods and there will be error, if child class doesn't implement them ), but not every Runner. I am not sure what is a correct approach here.

Reply:
remove abstract implementation from here, it's already abstract in `Runner` class